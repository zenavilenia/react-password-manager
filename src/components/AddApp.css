.add-app {
  float: right;
  background-color: #008CBA; /* Green */
  border: none;
  color: white;
  padding: 12px 28px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  border-radius: 8px;
  width: auto;
  margin: 10px;
}

.valid {
  color: green;
}

.invalid {
  color: red;
}

.left {
  float: left;
  padding: 20px;
}

.right {
  float: right;
}

/* Modal */
a {
  color: inherit;
}

.modal-window {
  position: fixed;
  background-color: rgba(255, 255, 255, 0.15);
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 999;
  opacity: 0;
  pointer-events: none;
  -webkit-transition: all 0.3s;
  -moz-transition: all 0.3s;
  transition: all 0.3s;
}

.modal-window:target {
  opacity: 1;
  pointer-events: auto;
}

.modal-window>div {
  width: 400px;
  position: relative;
  margin: 5% auto;
  padding: 1rem;
  background: #fff;
  color: #444;
}

.modal-window header {
  font-weight: bold;
}

.modal-close {
  color: #aaa;
  line-height: 50px;
  font-size: 80%;
  position: absolute;
  right: 0;
  text-align: center;
  top: 0;
  width: 70px;
  text-decoration: none;
}

.modal-close:hover {
  color: #000;
}

.modal-window h1 {
  font-size: 150%;
  margin: 0 0 15px;
}