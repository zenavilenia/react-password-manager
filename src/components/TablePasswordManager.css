#password-generator {
  font-family: "Trebuchet MS", Arial, Helvetica, sans-serif;
  border-collapse: collapse;
  width: 100%;
}

#password-generator td, #password-generator th {
  border: 1px solid #ddd;
  padding: 8px;
}

#password-generator tr:nth-child(even){background-color: #f2f2f2;}

#password-generator tr:hover {background-color: #ddd;}

#password-generator th {
  padding-top: 12px;
  padding-bottom: 12px;
  text-align: left;
  background-color: #white;
  color: black;
  border: 1px solid green;
}

.fa {
  padding: 3px;
}

/* Modal */
a {
  color: inherit;
}

.modal-window {
  position: fixed;
  background-color: rgba(255, 255, 255, 0.15);
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 999;
  opacity: 0;
  pointer-events: none;
  -webkit-transition: all 0.3s;
  -moz-transition: all 0.3s;
  transition: all 0.3s;
}

.modal-window:target {
  opacity: 1;
  pointer-events: auto;
}

.modal-window>div {
  width: 400px;
  position: relative;
  margin: 10% auto;
  padding: 2rem;
  background: #fff;
  color: #444;
}

.modal-window header {
  font-weight: bold;
}

.modal-close {
  color: #aaa;
  line-height: 50px;
  font-size: 80%;
  position: absolute;
  right: 0;
  text-align: center;
  top: 0;
  width: 70px;
  text-decoration: none;
}

.modal-close:hover {
  color: #000;
}

.modal-window h1 {
  font-size: 150%;
  margin: 0 0 15px;
}

/* Modal 2 */

.modal-edit-app-window {
  position: fixed;
  background-color: rgba(255, 255, 255, 0.15);
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 999;
  opacity: 0;
  pointer-events: none;
  -webkit-transition: all 0.3s;
  -moz-transition: all 0.3s;
  transition: all 0.3s;
}

.modal-edit-app-window:target {
  opacity: 1;
  pointer-events: auto;
}

.modal-edit-app-window>div {
  width: 400px;
  position: relative;
  margin: 10% auto;
  padding: 2rem;
  background: #fff;
  color: #444;
}

.modal-edit-app-window header {
  font-weight: bold;
}

.modal-edit-app-close {
  color: #aaa;
  line-height: 50px;
  font-size: 80%;
  position: absolute;
  right: 0;
  text-align: center;
  top: 0;
  width: 70px;
  text-decoration: none;
}

.modal-edit-app-close:hover {
  color: #000;
}

.modal-edit-app-window h1 {
  font-size: 150%;
  margin: 0 0 15px;
}