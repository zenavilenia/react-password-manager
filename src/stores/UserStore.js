import { observable, action, computed } from 'mobx';
import Firebase from 'firebase'

import {db} from '../firebase.js'

class UserStore {
  // to be implemented later on
  @observable user = {}
  @observable apps = []
  
  @action
  addApp = (app) => {
    const addApp = {
      ...app,
      createdAt: Firebase.database.ServerValue.TIMESTAMP,
      updatedAt: Firebase.database.ServerValue.TIMESTAMP,
    }
    const key = localStorage.getItem('key');
    db.ref(`users/${key}/apps`).push(addApp);
  };

  search = (searchkey) => {
    const key = localStorage.getItem('key');
    db.ref(`users/${key}/apps`).once('value', (snapshot) => {
      this.apps = []
      snapshot.forEach(element => {
        const app = { ...element.val(), key: element.key };
        app.showedPassword = '*'.repeat(app.password.length);
        if(app.app.includes(searchkey)) {
          this.apps.push(app);
        }
      })
    })
  }

  getAppList = () => {
    console.log('get app list')
    const key = localStorage.getItem('key');
    db.ref(`users/${key}/apps`).once('value', (snapshot) => {
      this.apps = []
      snapshot.forEach(element => {
        const app = { ...element.val(), key: element.key };
        app.showedPassword = '*'.repeat(app.password.length);
        this.apps.push(app);
      })
    })
  }

  checkPassword = (password, index) => {
    console.log('masuk check password')
    db.ref('users').once('value', (snapshot) => {
      snapshot.forEach(element => {
        const email = localStorage.getItem('email')
        const user = element.val();
        if (user.email === email && user.password === password) {
          console.log(element)
          this.showPassword(index);
        }
      })
    })
  }

  showPassword = (index) => {
    this.apps[index].showedPassword = this.apps[index].password;
  }

  editApp = (appkey, objApp) => {
    const key = localStorage.getItem('key');
    console.log('key', appkey)
    console.log('obk', objApp)
    objApp.updatedAt = Firebase.database.ServerValue.TIMESTAMP;
    db.ref(`users/${key}/apps/${appkey}`).set({ ...objApp })
  }

  deleteApp = (appkey) => {
    console.log('masuk')
    const key = localStorage.getItem('key');
    console.log(appkey)
    db.ref(`users/${key}/apps/${appkey}`).remove()
    this.getAppList();
  }

  login = (objUser) => {
    db.ref('users').once('value', (snapshot) => {
      snapshot.forEach(element => {
        const user = element.val();
        if (user.email === objUser.email && user.password === objUser.password) {
          console.log(element);
          localStorage.setItem('key', element.key);
          localStorage.setItem('email', user.email);
          this.user = { ...objUser };
        }
      })
    })
  }

  register = (objUser) => {
    console.log(objUser)

    const obj = {
      ...objUser,
      createdAt: Firebase.database.ServerValue.TIMESTAMP,
      updatedAt: Firebase.database.ServerValue.TIMESTAMP,
      apps: []
    }
    console.log('ini obj', obj)
    db.ref('users').push({...obj})

    this.login(objUser)
  }

  handleChange(that) {
    return function handleChangeInside(e) {
      that.setState({
        [e.target.name]: e.target.value,
      })
    }
  }

  handleChangePassword = (that) => {
    return function handleChangePasswordInside(e) {
      that.setState({
        [e.target.name]: e.target.value,
      }, () => {
        let uppercase = document.getElementById('uppercase');
        let lowercase = document.getElementById('lowercase');
        let specialcase = document.getElementById('specialcase');
        let number = document.getElementById('number');
        let minimaldigit = document.getElementById('minimaldigit');
        
        let upperCaseTest = /[A-Z]/g;
        if (that.state.password.match(upperCaseTest)) { 
          uppercase.classList = "valid";
        } else {
          uppercase.classList = "invalid";
        }
  
        let lowerCaseTest = /[a-z]/g;
        if (that.state.password.match(lowerCaseTest)) { 
          lowercase.classList = "valid";
        } else {
          lowercase.classList = "invalid";
        }
  
        let specialCaseTest = /[$&+,:;=?@#|'<>.^*()%!-]/g;
        if (that.state.password.match(specialCaseTest)) { 
          specialcase.classList = "valid";
        } else {
          specialcase.classList = "invalid";
        }
  
        let numberTest = /[0-9]/g;
        if (that.state.password.match(numberTest)) { 
          number.classList = "valid";
        } else {
          number.classList = "invalid";
        }
  
        let minimalDigitTest = /^.{6,}$/g;
        if (that.state.password.match(minimalDigitTest)) { 
          minimaldigit.classList = "valid";
        } else {
          minimaldigit.classList = "invalid";
        }
      })
    }
  }

  @computed
  get appCount() {
    return this.apps.length;
  }
}

export default new UserStore();