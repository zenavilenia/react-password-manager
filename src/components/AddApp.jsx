import React, { Component } from 'react';

import Search from './Search.jsx';
import UserStore from '../stores/UserStore';
import './AddApp.css';

class AddApp extends Component {
  constructor() {
    super();

    this.state = {
      app: '',
      username: '',
      password: '',
    };
  }

  addApp() {
    const newApp = {
      app: this.state.app,
      username: this.state.username,
      password: this.state.password,
    }
    UserStore.addApp({...newApp});
    UserStore.getAppList();
  }

  render() {
    return (
      <div>
        <div className="left">
          <Search/>
        </div>
        <div className="right">
          <button className="add-app">
            <a href="#open-modal">
              Add App
            </a>
          </button>
        </div>

        {/* Modal with css */}
        <div id="open-modal" class="modal-window">
          <div>
            <a href="#modal-close" title="Close" class="modal-close">Close</a>
            <form onSubmit={(e) => {this.addApp(); e.preventDefault();}}>
              <div className="container">
                <label><b>App</b></label>
                <input type="text" placeholder="App's name" name="app" id="appId" value={ this.state.app } onChange={ UserStore.handleChange(this) } required/>
                <label><b>Username</b></label>
                <input type="text" placeholder="Your username" name="username" id="usernameId" value={ this.state.username } onChange={ UserStore.handleChange(this) } required/>
                <label><b>Password</b></label>
                <input type="password" placeholder="Your password" name="password" id="passwordId" value={ this.state.password } onChange={ UserStore.handleChangePassword(this) } required/>
                <button type="submit">Add App</button>
              </div>
            </form>
            <hr/>
            <div>
              <p>Password Must Contain: </p>
              <div id="uppercase" className="invalid">Upper Case</div>
              <div id="lowercase" className="invalid">Lower Case</div>
              <div id="specialcase" className="invalid">Special Character</div>
              <div id="number" className="invalid">At least 1 number</div>
              <div id="minimaldigit" className="invalid">Minimal 6 Digits</div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default AddApp;