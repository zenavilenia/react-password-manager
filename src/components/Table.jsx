import React, { Component } from 'react';
import { inject, observer } from 'mobx-react';

import UserStore from '../stores/UserStore';

@inject('UserStore')
@observer class Table extends Component {
  render() {
    return (
      <table id="password-generator">
        <thead>
          <tr>
            <th>No</th>
            <th>App</th>
            <th>Username</th>
            <th>Password</th>
            <th>Created At</th>
            <th>Updated At</th>
            <th>Actions</th>
          </tr>
        </thead>
        <tbody>
          { UserStore.apps.map((app, i) => 
            <tr key={ app.key }>
              <th> { i+1 } </th>
              <th> { app.app } </th>
              <th> { app.username } </th>
              <th> { app.showedPassword } </th>
              <th> { this.props.that.dateFormat( app.createdAt ) } </th>
              <th> { this.props.that.dateFormat( app.updatedAt ) } </th>
              <th>
                <a onClick={ (e) => { this.props.that.changeIndex(i); }} href="#open-modal-password">
                  <i class="fa fa-eye"></i>
                </a> | 
                <a onClick={ (e) => { this.props.that.editState(app.key, app.app, app.username, app.password, app.createdAt); }} href="#open-modal-edit-app">
                <i class="fa fa-dot-circle-o"></i>
                </a> | 
                <a href="" onClick={ (e) => { UserStore.deleteApp(app.key); e.preventDefault(); }}>
                  <i class="fa fa-close"></i>
                </a>
              </th>
            </tr>
            ) }
        </tbody>
      </table>
    );
  }
}

export default Table;