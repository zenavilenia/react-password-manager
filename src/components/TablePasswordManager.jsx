import React, { Component } from 'react';
import { inject, observer } from 'mobx-react';

import AddApp from './AddApp.jsx';
import Table from './Table.jsx';
import UserStore from '../stores/UserStore';
import './TablePasswordManager.css'

@observer class TablePasswordManager extends Component {
  constructor() {
    super();
    this.state = {
      key: '',
      app: '',
      username: '',
      password: '',
      index: '',
      createdAt: '',
    }
  }

  componentDidMount() {
    UserStore.getAppList()
  }

  editState(key, app, username, password, createdAt) {
    this.setState({
      key, app, username, password, createdAt
    })
  }

  editApp() {
    const editThisApp = {
      app: this.state.app,
      username: this.state.username,
      password: this.state.password,
      createdAt: this.state.createdAt,
    }
    UserStore.editApp(this.state.key, {...editThisApp});
    UserStore.getAppList();
  }

  showPassword(appkey, pwd) {
    let app = document.getElementById(appkey);
    app.classList = "show-password";
    this.hidePassword(appkey, pwd);
  }

  checkOpenPwd = (pwd) => {
    UserStore.checkPassword(pwd, this.state.index)
  }

  changeIndex(i) {
    this.setState({
      index: i
    })
  }

  dateFormat (times) {
    let months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December']
    let date = new Date(times).getDate()
    let month = new Date(times).getMonth()
    let year = new Date(times).getFullYear()
    return (date < 10)? `0${date} ${months[month]} ${year}`: `${date} ${months[month]} ${year}`
  }

  render() {
    return (
      <div>
        <AddApp/>
        <Table that={this}/>

        {/* Modal with css*/}
        <div id="open-modal-password" class="modal-password-window">
          <div>
            <a href="#modal-password-close" title="Close" class="modal-close">Close</a>
            <div className="container">
              <label><b>Password</b></label>
              <input type="password" placeholder="Your password" name="password" value={ this.state.password } onChange={ UserStore.handleChange(this) } required/>
              <button type="button" onClick={() => {this.checkOpenPwd(this.state.password);}}>
                <a href="#modal-password-close" title="Close">Show Password</a>
              </button>
            </div>
          </div>
        </div>

        {/* Modal edit app*/}
        <div id="open-modal-edit-app" class="modal-edit-app-window">
          <div>
            <a href="#modal-edit-app-close" title="Close" class="modal-edit-app-close">Close</a>
            <div className="container">
              <label><b>App</b></label>
              <input type="text" placeholder="App's name" name="app" value={ this.state.app } onChange={ UserStore.handleChange(this) } required/>
              <label><b>Username</b></label>
              <input type="text" placeholder="Your username" name="username" value={ this.state.username } onChange={ UserStore.handleChange(this) } required/>
              <label><b>Password</b></label>
              <input type="password" placeholder="Your password" name="password" value={ this.state.password } onChange={ UserStore.handleChangePassword(this) } required/>
              <button type="button" onClick={() => {this.editApp();}}>
                <a href="#modal-edit-app-close" title="Close">Edit App</a>
              </button>
            </div>
            <hr/>
            <div>
              <p>Password Must Contain: </p>
              <div id="uppercase" className="invalid">Upper Case</div>
              <div id="lowercase" className="invalid">Lower Case</div>
              <div id="specialcase" className="invalid">Special Character</div>
              <div id="number" className="invalid">At least 1 number</div>
              <div id="minimaldigit" className="invalid">Minimal 6 Digits</div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default TablePasswordManager;