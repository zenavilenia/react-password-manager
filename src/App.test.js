import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import App from './App';
import Register from './components/Register';
import AddApp from './components/AddApp';
import Login from './components/Login';
import Navbar from './components/Navbar';
import TablePasswordManager from './components/TablePasswordManager'
import Search from './components/Search';
import LocalStorageMock from './LocalStorageMock.js';
import Table from './components/Table';
import UserStore from './stores/UserStore.js';

Enzyme.configure({ adapter: new Adapter() });

global.localStorage = new LocalStorageMock;

it('renders without crashing', () => {
  const div = document.createElement('div', { id: 'root' });
  ReactDOM.render(<div id="root"><App /></div>, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('tes the environment', () => {
  it('works, hopefully', () => {
    expect(true).toEqual(true)
  })
})

describe('<App/>', () => {
  it('should render <Navbar/>', () => {
    const wrapper = shallow(<App/>);
    expect(wrapper.containsAllMatchingElements([
      <Navbar/>
    ])).toBe(true)
  })
})

describe('<Register/>', () => {
  it('should render start with empty email', () => {
    const wrapper = shallow(<Register/>);
    expect(wrapper.state('email').length).toBe(0)
  })

  it('should render start with empty password', () => {
    const wrapper = shallow(<Register/>);
    expect(wrapper.state('password').length).toBe(0)
  })

  it('should have component form, div, button, input, label', () => {
    const component = renderer.create(<Register/>)
    expect(component).toMatchSnapshot();
    const wrapper = shallow(<Register/>);
    expect(wrapper.containsAllMatchingElements([
      <div/>,
      <form/>,
      <button/>,
      <input/>,
      <label/>
    ]))
    expect(component).toMatchSnapshot()
  })
})

describe('<Login/>', () => {
  it('should render start with empty email', () => {
    const wrapper = shallow(<Login/>);
    expect(wrapper.state('email').length).toBe(0)
  })

  it('should render start with empty password', () => {
    const wrapper = shallow(<Login/>);
    expect(wrapper.state('password').length).toBe(0)
  })

  it('should have component form, div, button, input, label', () => {
    const component = renderer.create(<Login/>)
    expect(component).toMatchSnapshot();
    const wrapper = shallow(<Login/>);
    expect(wrapper.containsAllMatchingElements([
      <div/>,
      <form/>,
      <button/>,
      <input/>,
      <label/>
    ]))
    expect(component).toMatchSnapshot()
  })
})

describe('<TablePasswordManager/>', () => {
  it('should render start with empty key', () => {
    const wrapper = shallow(<TablePasswordManager/>);
    expect(wrapper.state('key').length).toBe(0)
  })

  it('should render start with empty app', () => {
    const wrapper = shallow(<TablePasswordManager/>);
    expect(wrapper.state('app').length).toBe(0)
  })

  it('should render start with empty username', () => {
    const wrapper = shallow(<TablePasswordManager/>);
    expect(wrapper.state('username').length).toBe(0)
  })

  it('should render start with empty password', () => {
    const wrapper = shallow(<TablePasswordManager/>);
    expect(wrapper.state('password').length).toBe(0)
  })

  it('should render start with empty createdAt', () => {
    const wrapper = shallow(<TablePasswordManager/>);
    expect(wrapper.state('createdAt').length).toBe(0)
  })

  it('should render start with empty index', () => {
    const wrapper = shallow(<TablePasswordManager/>);
    expect(wrapper.state('index').length).toBe(0)
  })

  it('should render <AddApp/>', () => {
    const wrapper = shallow(<TablePasswordManager/>);
    expect(wrapper.containsAllMatchingElements([
      <AddApp/>
    ])).toBe(true)
  })

  it('should render <Table/>', () => {
    const wrapper = shallow(<TablePasswordManager/>);
    expect(wrapper.containsAllMatchingElements([
      <Table/>
    ])).toBe(true)
  })
})

describe('<AddApp/>', () => {
  it('should render start with empty app', () => {
    const wrapper = shallow(<AddApp/>);
    expect(wrapper.state('app').length).toBe(0)
  })

  it('should render start with empty username', () => {
    const wrapper = shallow(<AddApp/>);
    expect(wrapper.state('username').length).toBe(0)
  })

  it('should render start with empty password', () => {
    const wrapper = shallow(<AddApp/>);
    expect(wrapper.state('password').length).toBe(0)
  })

  it('should render <Search/>', () => {
    const wrapper = shallow(<AddApp/>);
    expect(wrapper.containsAllMatchingElements([
      <Search/>
    ])).toBe(true)
  })

  it('should add new app', async () => {
    const wrapper = shallow(<AddApp/>);
    wrapper.setState({
      app: 'testapp',
      username: 'testusername',
      password: 'testpassword'
    });
    wrapper.find('form')
           .simulate('submit', {
             preventDefault () { }
           })
    
  })
})

describe('<Search/>', () => {
  it('should render start with empty app', () => {
    const wrapper = shallow(<Search/>);
    expect(wrapper.state('search').length).toBe(0)
  })
})

// describe('UserStore', () => {
//   it('should create new app', () => {
//     const store = new UserStore;
    
//   })
// })