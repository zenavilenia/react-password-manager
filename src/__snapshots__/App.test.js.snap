// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> should have component form, div, button, input, label 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="container"
    >
      <label>
        <b>
          Email
        </b>
      </label>
      <input
        name="email"
        onChange={[Function]}
        placeholder="Enter your email"
        required={true}
        type="text"
        value=""
      />
      <label>
        <b>
          Password
        </b>
      </label>
      <input
        name="password"
        onChange={[Function]}
        placeholder="Enter Password"
        required={true}
        type="password"
        value=""
      />
      <button
        type="submit"
      >
        Login
      </button>
      <hr />
      <a
        href=""
        onClick={[Function]}
      >
        Don't have account? Register here
      </a>
    </div>
  </form>
</div>
`;

exports[`<Login/> should have component form, div, button, input, label 2`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="container"
    >
      <label>
        <b>
          Email
        </b>
      </label>
      <input
        name="email"
        onChange={[Function]}
        placeholder="Enter your email"
        required={true}
        type="text"
        value=""
      />
      <label>
        <b>
          Password
        </b>
      </label>
      <input
        name="password"
        onChange={[Function]}
        placeholder="Enter Password"
        required={true}
        type="password"
        value=""
      />
      <button
        type="submit"
      >
        Login
      </button>
      <hr />
      <a
        href=""
        onClick={[Function]}
      >
        Don't have account? Register here
      </a>
    </div>
  </form>
</div>
`;

exports[`<Register/> should have component form, div, button, input, label 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="container"
    >
      <label>
        <b>
          Email
        </b>
      </label>
      <input
        name="email"
        onChange={[Function]}
        placeholder="Enter your email"
        required={true}
        type="text"
        value=""
      />
      <label>
        <b>
          Password
        </b>
      </label>
      <input
        name="password"
        onChange={[Function]}
        placeholder="Enter Password"
        required={true}
        type="password"
        value=""
      />
      <button
        type="submit"
      >
        Register
      </button>
      <hr />
      <a
        href=""
        onClick={[Function]}
      >
        Already have account? Login here
      </a>
    </div>
  </form>
</div>
`;

exports[`<Register/> should have component form, div, button, input, label 2`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="container"
    >
      <label>
        <b>
          Email
        </b>
      </label>
      <input
        name="email"
        onChange={[Function]}
        placeholder="Enter your email"
        required={true}
        type="text"
        value=""
      />
      <label>
        <b>
          Password
        </b>
      </label>
      <input
        name="password"
        onChange={[Function]}
        placeholder="Enter Password"
        required={true}
        type="password"
        value=""
      />
      <button
        type="submit"
      >
        Register
      </button>
      <hr />
      <a
        href=""
        onClick={[Function]}
      >
        Already have account? Login here
      </a>
    </div>
  </form>
</div>
`;
